### DISCUSSION TITLE
Start to work with Vanjaro

### DISCUSSION THREAD

**Discussion by maduranga001** [vanjarosoftware/Vanjaro.Platform#1576] [Q&A]
Hi,

I checked the Vanjaro and learning the framework. I have a few questions and would appreciate it if you can help me with this.

1) Can I set up and compile the Vanjaro Source code to do any changes to the back-end? Please, can you advise me on how to do it. I just want to prepare the source version for the modifications.

2) What are the best tools to use for the backend modifications on Vanjaro source code? (Visual Studio version and MSSQL version)

3) Is the latest Vanjaro version working with the latest DNN 9.10.0 version?

4) How can I add new custom Block + Page Templates to the template library?

5) How can I add a new Basic/Design/App to the Blocks?

I would appreciate it if you can support me to start with this.

**Comment by meetmandeep** [comment]
> Can I set up and compile the Vanjaro Source code to do any changes to the back-end? Please, can you advise me on how to do it. I just want to prepare the source version for the modifications.

Yes. We'll have a webinar soon on setting up a dev environment. 

But in a nutshell: 

1. Create a Vanjaro Site on your local machine. Refer to Vanjaro Installer in Wiki. 
2. Fork Vanjaro.Platform repo and clone on same folder as Vanjaro Site

> What are the best tools to use for the backend modifications on Vanjaro source code? (Visual Studio version and MSSQL version)

- Visual Studio 2019 (Community Edition is fine)
- SQL Server 2019 Developer Edition (But Express works as well)

> Is the latest Vanjaro version working with the latest DNN 9.10.0 version?

The current Vanjaro Platform is built on DNN 9.9.1. Vanjaro for DNN Extension is compatible with DNN 9.9.1+ including DNN 9.10.0

> How can I add new custom Block + Page Templates to the template library?

Currently, Template library is only available to Premium Theme users. Premium Theme will be available for purchase in Vanjaro App Store (Coming Soon). 3rd party vendors will be able to sell their own themes and add their own templates (pages / blocks / sites) to template library. 

For now, you may create your own custom blocks and save them locally under the Custom Blocks tab. 

> How can I add a new Basic/Design/App to the Blocks?

Vanjaro supports DNN Modules; so you may create and install an app similar to DNN Module. In addition, Vanjaro supports new extension points for Blocks and we'll cover these in depth in upcoming webinars.

**Reply by maduranga001** [reply]
Please, can you explain properly how to set up the module source code for backend changes and compile it? really appreciate if you can advise on this. Better to have a written answer until we have the webinar.

**Reply by meetmandeep** [reply]
See: https://github.com/vanjarosoftware/Vanjaro.Platform/discussions/1623#discussioncomment-1197014
